modules/dnn/src/op_inf_engine.cpp

```diff
@@ -651,6 +651,18 @@ InferenceEngine::Core& getCore(const std::string& id)
 }
 #endif
 
+static bool detectArmPlugin_()
+{
+#if INF_ENGINE_VER_MAJOR_GT(INF_ENGINE_RELEASE_2021_2) && (defined(__arm__) || defined(__aarch64__))
+    AutoLock lock(getInitializationMutex());
+    InferenceEngine::Core& ie = getCore("CPU");
+    const std::vector<std::string> devices = ie.GetAvailableDevices();
+    return std::find(devices.begin(), devices.end(), std::string("CPU")) != devices.end();
```

Comment:
@ilya-lavrenov Is there some robust approach to detect plugin type through InferenceEngine QueryAPI?

AFAIK, for example, GNA has software and hardware-accelerated implementations.

Reply:
@l-bat Please use `FULL_DEVICE_NAME` and check for `"arm_compute::NEON"`:

https://github.com/openvinotoolkit/openvino_contrib/blob/9acb635162d0ff0cf35cad780f668a9e2b4cdece/modules/arm_plugin/src/arm_plugin.cpp#L187-L188

Reply:
`FULL_DEVICE_NAME` for ARM plugin has `"arm_compute::NEON"`. Maybe we can rely on it..